File: packages/twilio_flutter_conversations/ios/Classes/TwilioFlutterConversationsPlugin.swift
import Flutter
import UIKit

public class TwilioFlutterConversationsPlugin: NSObject {
  private var bridge: ConversationsBridge?

  public static func register(with registrar: FlutterPluginRegistrar) {
    let bridge = ConversationsBridge(binaryMessenger: registrar.messenger())
    let hostApi = ConversationsBridgeHostAdapter(bridge: bridge)
    TwilioConversationsHostApiSetup.setUp(
      binaryMessenger: registrar.messenger(),
      api: hostApi
    )

    let instance = TwilioFlutterConversationsPlugin()
    instance.bridge = bridge
  }

  deinit {
    bridge?.dispose()
  }
}

File: packages/twilio_flutter_conversations/android/src/main/kotlin/com/twilioflutter/conversations/TwilioFlutterConversationsPlugin.kt
package com.twilioflutter.conversations

import com.twilioflutter.conversations.bridge.ConversationsBridge
import com.twilioflutter.conversations.bridge.ConversationsBridgeHostAdapter
import com.twilioflutter.conversations.pigeon.TwilioConversationsHostApi
import io.flutter.embedding.engine.plugins.FlutterPlugin

/** Flutter plugin entry point for Twilio Conversations. */
class TwilioFlutterConversationsPlugin : FlutterPlugin {
  private var bridge: ConversationsBridge? = null

  override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    val conversationsBridge =
      ConversationsBridge(
        context = binding.applicationContext,
        binaryMessenger = binding.binaryMessenger,
      )
    bridge = conversationsBridge
    TwilioConversationsHostApi.setUp(
      binding.binaryMessenger,
      ConversationsBridgeHostAdapter(conversationsBridge),
    )
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    bridge?.dispose()
    bridge = null
    TwilioConversationsHostApi.setUp(binding.binaryMessenger, null)
  }
}
